Web Inspector: Layout sidebar in Elements tab lacks an discernable visual hierarchy

dcrousso opened the pull request:
#### bc5033fed4cfe5d60ce3a3bc0a9d48edb68d574a
<pre>
Web Inspector: Layout sidebar in Elements tab lacks an discernable visual hierarchy
<a href="https://bugs.webkit.org/show_bug.cgi?id=239525">https://bugs.webkit.org/show_bug.cgi?id=239525</a>
&lt;rdar://problem/91997854&gt;

Reviewed by Patrick Angle.

* UserInterface/Views/LayoutDetailsSidebarPanel.js:
(WI.LayoutDetailsSidebarPanel.prototype.initialLayout):
(WI.LayoutDetailsSidebarPanel.prototype.layout):
* UserInterface/Views/LayoutDetailsSidebarPanel.css:
(.details-section:is(.layout-css-flexbox, .layout-css-grid) &gt; .content &gt; .group &gt; .row): Added.
(.details-section:is(.layout-css-flexbox, .layout-css-grid) &gt; .content &gt; .group &gt; .row.empty): Added.
(.details-section:is(.layout-css-flexbox, .layout-css-grid) &gt; .content .settings-group &gt; .title): Added.
(.details-section:is(.layout-css-flexbox, .layout-css-grid) &gt; .content .setting-editor input[type=&quot;checkbox&quot;]): Added.
(.node-overlay-list-section &gt; h2 &gt; .options): Added.
(.node-overlay-list-section:not(:hover) &gt; h2 &gt; .options): Added.
Move the UI for each overlay&apos;s settings to be in the panel, not in the `WI.NodeOverlayListSection`.

* UserInterface/Views/NodeOverlayListSection.js:
(WI.NodeOverlayListSection):
(WI.NodeOverlayListSection.prototype.initialLayout):
(WI.NodeOverlayListSection.prototype.layout):
(WI.NodeOverlayListSection.prototype._handleOverlayStateChanged):
(WI.NodeOverlayListSection.prototype.get sectionLabel): Deleted.
(WI.NodeOverlayListSection.prototype._handleToggleAllCheckboxChanged): Deleted.
(WI.NodeOverlayListSection.prototype._updateToggleAllCheckbox): Deleted.
* UserInterface/Views/NodeOverlayListSection.css:
(.node-overlay-list-section &gt; h2): Added.
(.node-overlay-list-section .node-overlay-list-item-container input[type=&quot;checkbox&quot;]): Added.
(.node-overlay-list-section): Deleted.
(.node-overlay-list-section &gt; .heading,): Deleted.
(.node-overlay-list-section &gt; .heading &gt; label &gt; .toggle-all): Deleted.
(.node-overlay-list-section :is(.setting-editor, .node-overlay-list-item-container, .heading) input[type=&quot;checkbox&quot;]): Deleted.
* UserInterface/Views/CSSFlexNodeOverlayListSection.js: Removed.
* UserInterface/Views/CSSGridNodeOverlayListSection.js: Removed.
Remove the subclasses now that the only differentiating factor between them is the label.
Also remove the checkbox in the heading that enables/disables all overlays because it&apos;s
visually odd (i.e. the header now lines up with it&apos;s content, instead of being less indented)
and because there&apos;s likely little use case in enabling *all* overlays at once.

* UserInterface/Views/SettingsGroup.js:
(WI.SettingsGroup):
* UserInterface/Views/SettingsView.js:
(WI.SettingsView):
(WI.SettingsView.prototype.addCenteredContainer):
* UserInterface/Views/SettingsTabContentView.css:
(.content-view.tab.settings &gt; .settings-view &gt; :is(.settings-group, .container)): Renamed from `.content-view.tab.settings &gt; .settings-view &gt; .container`.
(.content-view.tab.settings &gt; .settings-view &gt; .container.centered): Renamed from `.content-view.tab.settings &gt; .settings-view &gt; .container-centered`.
(.content-view.tab.settings &gt; .settings-view &gt; .settings-group &gt; .title): Renamed from `.content-view.tab.settings &gt; .settings-view &gt; .container &gt; .title`.
(.content-view.tab.settings &gt; .settings-view &gt; .settings-group &gt; .editor-group): Renamed from `.content-view.tab.settings &gt; .settings-view &gt; .container &gt; .editor-group`.
(.content-view.tab.settings &gt; .settings-view &gt; .settings-group &gt; .editor-group &gt; .setting-editor): Renamed from `.content-view.tab.settings &gt; .settings-view &gt; .container &gt; .editor-group &gt; .setting-editor`.
(.content-view.tab.settings &gt; .settings-view &gt; .settings-group &gt; .editor-group &gt; .setting-editor:first-child &gt; *): Renamed from `.content-view.tab.settings &gt; .settings-view &gt; .container &gt; .editor-group &gt; .setting-editor:first-child &gt; *`.
(.content-view.tab.settings &gt; .settings-view &gt; .settings-group &gt; .editor-group &gt; .setting-editor input[type=&quot;checkbox&quot;]): Renamed from `.content-view.tab.settings &gt; .settings-view &gt; .container &gt; .editor-group &gt; .setting-editor input[type=&quot;checkbox&quot;]`.
(.content-view.tab.settings &gt; .settings-view &gt; .settings-group &gt; .editor-group &gt; .setting-editor select): Renamed from `.content-view.tab.settings &gt; .settings-view &gt; .container &gt; .editor-group &gt; .setting-editor select`.
(.content-view.tab.settings &gt; .settings-view &gt; .settings-group &gt; .editor-group &gt; .setting-editor input[type=&quot;number&quot;]): Renamed from `.content-view.tab.settings &gt; .settings-view &gt; .container &gt; .editor-group &gt; .setting-editor input[type=&quot;number&quot;]`.
(.content-view.tab.settings &gt; .settings-view &gt; .settings-group &gt; .editor-group &gt; .setting-editor input[type=&quot;text&quot;]): Renamed from `.content-view.tab.settings &gt; .settings-view &gt; .container &gt; .editor-group &gt; .setting-editor input[type=&quot;text&quot;]`.
Use more descriptive/specific CSS classes.

* UserInterface/Views/BlackboxSettingsView.css:
(.settings-view.blackbox &gt; p &gt; label &gt; input[type=&quot;checkbox&quot;]):
Drive-by: Make the checkbox in the Blackbox pane the same size as those in every other pane.

* UserInterface/Views/DetailsSection.css:
(.details-section &gt; .content &gt; .group:nth-child(even) &gt; .row:matches(.simple:first-child &gt; *, :not(.simple):first-child)): Deleted.
Drive-by: This ruleset caused the border between two `WI.DetailsSectionGroup` to be thicker
          than it should be. AFAICT no other `WI.DetailsSectionGroup` even use this ruleset
          because they contain `WI.DetailsSectionSimpleRow`.

* UserInterface/Main.html:
* Source/WebInspectorUI/Localizations/en.lproj/localizedStrings.js:

Canonical link: <a href="https://commits.webkit.org/254637@main">https://commits.webkit.org/254637@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/95dba1af6cadfa2fb13c1c041ac47d155c7a9681

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/89742 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/34293 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/20455 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/99087 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/155903 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/32787 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/28250 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/82111 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/93437 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/95389 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/68/builds/26056 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/76587 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/82111 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/80933 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/20455 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/82111 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/66/builds/30539 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/20455 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/67/builds/30288 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/20455 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/33739 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/76587 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/1373 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/32449 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/20455 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

patrickangle reviewed (approved):
I'm still a little hesitant about hiding the "Show/Hide all" checkboxes away behind a menu, but I'm willing to try it, because the rest of this looks soooooooooo much better.

dcrousso committed:
Web Inspector: Layout sidebar in Elements tab lacks an discernable visual hierarchy
https://bugs.webkit.org/show_bug.cgi?id=239525
<rdar://problem/91997854>

Reviewed by Patrick Angle.

* UserInterface/Views/LayoutDetailsSidebarPanel.js:
(WI.LayoutDetailsSidebarPanel.prototype.initialLayout):
(WI.LayoutDetailsSidebarPanel.prototype.layout):
* UserInterface/Views/LayoutDetailsSidebarPanel.css:
(.details-section:is(.layout-css-flexbox, .layout-css-grid) > .content > .group > .row): Added.
(.details-section:is(.layout-css-flexbox, .layout-css-grid) > .content > .group > .row.empty): Added.
(.details-section:is(.layout-css-flexbox, .layout-css-grid) > .content .settings-group > .title): Added.
(.details-section:is(.layout-css-flexbox, .layout-css-grid) > .content .setting-editor input[type="checkbox"]): Added.
(.node-overlay-list-section > h2 > .options): Added.
(.node-overlay-list-section:not(:hover) > h2 > .options): Added.
Move the UI for each overlay's settings to be in the panel, not in the `WI.NodeOverlayListSection`.

* UserInterface/Views/NodeOverlayListSection.js:
(WI.NodeOverlayListSection):
(WI.NodeOverlayListSection.prototype.initialLayout):
(WI.NodeOverlayListSection.prototype.layout):
(WI.NodeOverlayListSection.prototype._handleOverlayStateChanged):
(WI.NodeOverlayListSection.prototype.get sectionLabel): Deleted.
(WI.NodeOverlayListSection.prototype._handleToggleAllCheckboxChanged): Deleted.
(WI.NodeOverlayListSection.prototype._updateToggleAllCheckbox): Deleted.
* UserInterface/Views/NodeOverlayListSection.css:
(.node-overlay-list-section > h2): Added.
(.node-overlay-list-section .node-overlay-list-item-container input[type="checkbox"]): Added.
(.node-overlay-list-section): Deleted.
(.node-overlay-list-section > .heading,): Deleted.
(.node-overlay-list-section > .heading > label > .toggle-all): Deleted.
(.node-overlay-list-section :is(.setting-editor, .node-overlay-list-item-container, .heading) input[type="checkbox"]): Deleted.
* UserInterface/Views/CSSFlexNodeOverlayListSection.js: Removed.
* UserInterface/Views/CSSGridNodeOverlayListSection.js: Removed.
Remove the subclasses now that the only differentiating factor between them is the label.
Also remove the checkbox in the heading that enables/disables all overlays because it's
visually odd (i.e. the header now lines up with it's content, instead of being less indented)
and because there's likely little use case in enabling *all* overlays at once.

* UserInterface/Views/SettingsGroup.js:
(WI.SettingsGroup):
* UserInterface/Views/SettingsView.js:
(WI.SettingsView):
(WI.SettingsView.prototype.addCenteredContainer):
* UserInterface/Views/SettingsTabContentView.css:
(.content-view.tab.settings > .settings-view > :is(.settings-group, .container)): Renamed from `.content-view.tab.settings > .settings-view > .container`.
(.content-view.tab.settings > .settings-view > .container.centered): Renamed from `.content-view.tab.settings > .settings-view > .container-centered`.
(.content-view.tab.settings > .settings-view > .settings-group > .title): Renamed from `.content-view.tab.settings > .settings-view > .container > .title`.
(.content-view.tab.settings > .settings-view > .settings-group > .editor-group): Renamed from `.content-view.tab.settings > .settings-view > .container > .editor-group`.
(.content-view.tab.settings > .settings-view > .settings-group > .editor-group > .setting-editor): Renamed from `.content-view.tab.settings > .settings-view > .container > .editor-group > .setting-editor`.
(.content-view.tab.settings > .settings-view > .settings-group > .editor-group > .setting-editor:first-child > *): Renamed from `.content-view.tab.settings > .settings-view > .container > .editor-group > .setting-editor:first-child > *`.
(.content-view.tab.settings > .settings-view > .settings-group > .editor-group > .setting-editor input[type="checkbox"]): Renamed from `.content-view.tab.settings > .settings-view > .container > .editor-group > .setting-editor input[type="checkbox"]`.
(.content-view.tab.settings > .settings-view > .settings-group > .editor-group > .setting-editor select): Renamed from `.content-view.tab.settings > .settings-view > .container > .editor-group > .setting-editor select`.
(.content-view.tab.settings > .settings-view > .settings-group > .editor-group > .setting-editor input[type="number"]): Renamed from `.content-view.tab.settings > .settings-view > .container > .editor-group > .setting-editor input[type="number"]`.
(.content-view.tab.settings > .settings-view > .settings-group > .editor-group > .setting-editor input[type="text"]): Renamed from `.content-view.tab.settings > .settings-view > .container > .editor-group > .setting-editor input[type="text"]`.
Use more descriptive/specific CSS classes.

* UserInterface/Views/BlackboxSettingsView.css:
(.settings-view.blackbox > p > label > input[type="checkbox"]):
Drive-by: Make the checkbox in the Blackbox pane the same size as those in every other pane.

* UserInterface/Views/DetailsSection.css:
(.details-section > .content > .group:nth-child(even) > .row:matches(.simple:first-child > *, :not(.simple):first-child)): Deleted.
Drive-by: This ruleset caused the border between two `WI.DetailsSectionGroup` to be thicker
          than it should be. AFAICT no other `WI.DetailsSectionGroup` even use this ruleset
          because they contain `WI.DetailsSectionSimpleRow`.

* UserInterface/Main.html:
* Source/WebInspectorUI/Localizations/en.lproj/localizedStrings.js:

Canonical link: https://commits.webkit.org/254637@main
